src/java/org/apache/cassandra/cql3/constraints/ColumnConstraint.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.cql3.constraints;
+
+import org.apache.cassandra.cql3.CqlBuilder;
+import org.apache.cassandra.db.marshal.AbstractType;
+import org.apache.cassandra.io.IVersionedSerializer;
+import org.apache.cassandra.schema.ColumnMetadata;
+
+/**
+ * Common class for the conditions that a CQL Constraint needs to implement to be integrated in the
+ * CQL Constraints framework.
+ */
+public interface ColumnConstraint
+{
+    IVersionedSerializer<ColumnConstraint> serializer();
+
+    void appendCqlTo(CqlBuilder builder);
+
+    /**
+     * Method that evaluates the condition. It can either succeed or throw a {@link ConstraintViolationException}.
+     *
+     * @param valueType value type of the column value under test
+     * @param columnValue Column value to be evaluated at write time
+     */
+    void evaluate(Class<? extends AbstractType> valueType, Object columnValue) throws ConstraintViolationException;
```

Comment:
Could not we use javax.validation dependency / jar and use exceptions from there instead of writing our own? We already depend on javax.inject jar.

Reply:
I think there is value on specific exceptions for constraints. Downstream, we could decide to catch those exceptions differently, log them differently, grab some stats... This is not a hill I will die on, so if you really think there is no value in new exceptions I can go with the standard validation one. But, my preference would be to keep them.